mmv1/third_party/terraform/tests/resource_redis_instance_test.go

```diff
@@ -40,6 +40,151 @@ func TestAccRedisInstance_update(t *testing.T) {
 	})
 }
 
+// Validate that read replica is enabled on the instance without having to recreate
+func TestAccRedisInstance_updateReadReplicasMode(t *testing.T) {
+	t.Parallel()
+
+	name := fmt.Sprintf("tf-test-%d", randInt(t))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckRedisInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRedisInstanceReadReplicasUnspecified(name, true),
+			},
+			{
+				ResourceName:      "google_redis_instance.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccRedisInstanceReadReplicasEnabled(name, true),
+			},
+			{
+				ResourceName:      "google_redis_instance.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccRedisInstanceReadReplicasUnspecified(name, false),
+			},
+		},
+	})
+}
+
+/* Validate that read replica is enabled on the instance without recreate
+ * and secondaryIp is auto provisioned when passed as 'auto' */
+func TestAccRedisInstance_updateReadReplicasModeWithAutoSecondaryIp(t *testing.T) {
+	t.Parallel()
+
+	name := fmt.Sprintf("tf-test-%d", randInt(t))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckRedisInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRedisInstanceReadReplicasUnspecified(name, true),
+			},
+			{
+				ResourceName:      "google_redis_instance.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccRedisInstanceReadReplicasEnabledWithAutoSecondaryIp(name, true),
+			},
+			{
+				ResourceName:      "google_redis_instance.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccRedisInstanceReadReplicasUnspecified(name, false),
```

Comment:
Should this also disallow recreate?

Reply:
preventDestroy is turned off to allow cleanup o. f the resource. This part is not actually testing anything